testing/agent-for-testing/src/test/java/io/opentelemetry/javaagent/testing/AgentForTestingTest.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.testing;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.javaagent.testing.common.AgentTestingExporterAccess;
+import io.opentelemetry.sdk.trace.data.SpanData;
+import java.util.List;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class AgentForTestingTest {
+
+  @BeforeEach
+  void reset() {
+    AgentTestingExporterAccess.reset();
+  }
+
+  @Test
+  void empty() {
+    assertEquals(0, AgentTestingExporterAccess.getExportedSpans().size());
+  }
+
+  @Test
+  void exportAndRetrieve() {
```

Comment:
Goes through instrumentation API bridge, comes back from the SDK exporter through the agent classloader access

Reply:
👏